clang/test/CodeGen/builtins-nvptx.c

```diff
@@ -574,6 +578,12 @@ __device__ void nvvm_atom(float *fp, float f, double *dfp, double df, int *ip,
   __nvvm_atom_sys_cas_gen_ll(&sll, ll, 0);
 #endif
 
+#if __CUDA_ARCH__ >= 700
+  // CHECK_PTX63_SM70: cmpxchg ptr {{.*}} seq_cst seq_cst, align 2
+  // CHECK_PTX63_SM70-NEXT: extractvalue { i16, i1 } {{%[0-9]+}}, 0
+  __nvvm_atom_cas_gen_us(usp, 0, us);
```

Comment:
I think `__nvvm_atom_sys_cas_gen_us` and `__nvvm_atom_cta_cas_gen_us` also need to be tested.